pkg/core/consensus/user/provisioners.go

```diff
@@ -63,12 +63,18 @@ func NewProvisioners() *Provisioners {
 	}
 }
 
-// SizeAt returns the amount of provisioners active on a given round.
-func (p Provisioners) SizeAt(round uint64) int {
+// NumActive returns how many provisioners are active on a given round.
+// This function is used to determine the correct committee size for
+// sortition in the case where one or more provisioner stakes have not
+// yet become active, or have just expired. Note that this function will
+// only give an accurate result if the round given is either identical
+// or close to the current block height, as stakes are removed soon
+// after they expire.
+func (p Provisioners) NumActive(round uint64) int {
 	var size int
 	for _, member := range p.Members {
 		for _, stake := range member.Stakes {
-			if stake.StartHeight <= round {
+			if stake.StartHeight <= round && stake.EndHeight >= round {
```

Comment:
Nitpicking: it would be best to stick as closely as possible to the common mathematical notation for inequalities: `a <= x <= b`. In this case it would be best to rewrite the inequality as follows:

```
if stake.StartHeight <= round && round <= stake.EndHeight